### PR TITLE
feat: update event message to match CloudEvents spec

### PR DESCRIPTION
**Pull request by jyecusch**
Not fully compatible with cloud events yet, but closer. This shouldn't be a breaking change.

### CODE REVIEW

File: contracts/proto/event/v1/event.proto [hunk 1]
```diff
@@ -63,7 +63,11 @@ message NitricEvent {
...
   // A content hint for the events payload
-  string payload_type = 2;
+  string type = 2;
   // The payload of the event
-  google.protobuf.Struct payload = 3;
+  oneof data {
+    bytes binary_data = 4;
+    string text_data = 5;
+    google.protobuf.Struct proto_data = 3;
+  }
```
Comment (asalkeld) [inline contracts/proto/event/v1/event.proto:72]: if you wanted to be backwards compatible you could just have all fields, then just use the populated data

File: contracts/proto/event/v1/event.proto [hunk 2]
```diff
@@ -63,7 +63,11 @@ message NitricEvent {
...
   string id = 1;
   // A content hint for the events payload
-  string payload_type = 2;
+  string type = 2;
   // The payload of the event
-  google.protobuf.Struct payload = 3;
+  oneof data {
+    bytes binary_data = 4;
+    string text_data = 5;
+    google.protobuf.Struct proto_data = 3;
```
Comment (tjholm) [inline contracts/proto/event/v1/event.proto:71]: maybe change this to struct_data or object_data?

Comment (jyecusch) [reply contracts/proto/event/v1/event.proto:71]: I took the name from the CloudEvents proto spec for consistency https://github.com/cloudevents/spec/blob/v1.0.1/protobuf-format.md#3-data maybe we should just use this

**Review by tjholm — APPROVED** [review]

**Review by tjholm — CHANGES_REQUESTED** [review]
need to generate new server/client code and update handlers.

**Pull request closed without merging**